iac/terraform/aws/auth_cognito/cognito.tf

```diff
@@ -0,0 +1,69 @@
+resource "aws_cognito_user_pool" "user_pool" {
+  name = var.user_pool_name
+
+  schema {
+    attribute_data_type = "String"
+    name                = "email"
+    required            = true
+    mutable             = false
+  }
+
+  schema {
+    attribute_data_type = "String"
+    name                = "name"
+    required            = true
+    mutable             = true
+  }
+
+  schema {
+    attribute_data_type = "String"
+    name                = "phone_number"
+    required            = true
+    mutable             = true
+  }
+
+  auto_verified_attributes = ["email"]
+  username_attributes      = ["email"]
+
+  lambda_config {
+    custom_message    = module.lambda["custom_email_template"].lambda_arn
+    post_confirmation = module.lambda["post_confirmation"].lambda_arn
+  }
+
+  password_policy {
+    minimum_length    = 10
+    require_lowercase = true
+    require_numbers   = true
+    require_uppercase = false
+    require_symbols   = true
+  }
+
+  admin_create_user_config {
+    allow_admin_create_user_only = false
+  }
+
+  verification_message_template {
+    default_email_option = "CONFIRM_WITH_CODE"
+    email_message        = "Your verification code is {####}."
+    email_subject        = "Your Verification Code"
+  }
+  mfa_configuration = "OFF"
+  email_configuration {
+    email_sending_account = "DEVELOPER"
+    from_email_address    = "no-reply@${var.domain_name}"
```

Comment:
Full email address should be passed from terraform variable

Reply:
I wasn't sure if I should hardcode this in the variable, but I wanted to make it dynamic, which is why I put it in local to format the email. If there's a better option, please let me know, and I'll update it accordingly. [Commit](https://github.com/CraftsmenLtd/BloodConnect/commit/28f045306270fdc86a03546bd3b3e08356c4dc74#diff-de0ee50e8abad6227b92d1269d6b63da2edc830def52ba433c7b727cc65c9bac)

Reply:
Pass Email address via variable. You will find this in this [commit](https://github.com/CraftsmenLtd/BloodConnect/pull/113/commits/67d02235a3c4d0aeef23aeada28a414555c9de6c)